### PR TITLE
Make gracefulPeriod configurable

### LINKED ISSUE #2353: Shutting down tempo always takes at least 30 seconds
**Describe the bug**

Shutting down the tempo service takes 30 seconds. This is annoying when rebooting because it holds up the reboot. I assume it's because [of this code in the distributor](https://github.com/grafana/tempo/blob/469c3337763d7ce75bdcdf74dce0730932753b3e/modules/distributor/receiver/shim.go#L253-L278) that just sleeps for 30 seconds

**To Reproduce**

1. Run tempo
2. SIGTERM tempo
3. It takes 30 seconds

**Expected behavior**

It should shut down when all requests have been properly processed

**Environment:**

I'm running tempo in a systemd service on NixOS, the configuration for it can be found here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/tracing/tempo/default.nix

```
tempo, version 2.0.0 (branch: <release>, revision: 2.0.0)
  build user:
  build date:
  go version:       go1.19.5
  platform:         linux/amd64
```

### PR DESCRIPTION
**Pull request by Joffref**
**What this PR does**:

As mentioned it makes the graceful period configurable for ring distributor.

**Which issue(s) this PR fixes**:
Fixes #2353

**Checklist**
- [ ] Tests updated - Don't know if this relevant
- [ ] Documentation added - Don't know if this relevant
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## main / unreleased
 
+* [FEATURE] Make ring distributor gracefulPeriod configurable [#2604](https://github.com/grafana/tempo/pull/2604) (@joffref)
```
Comment (mapno) [inline CHANGELOG.md:3]:
```suggestion
* [ENHANCEMENT] Make ring distributor gracefulPeriod configurable [#2604](https://github.com/grafana/tempo/pull/2604) (@joffref)
```

It seems more like and enhancement than a feature to me—it's not really adding a new functionality to Tempo.

File: modules/distributor/distributor_ring.go [hunk 2]
```diff
@@ -21,6 +21,7 @@ type RingConfig struct {
 	KVStore          kv.Config     `yaml:"kvstore"`
 	HeartbeatPeriod  time.Duration `yaml:"heartbeat_period"`
 	HeartbeatTimeout time.Duration `yaml:"heartbeat_timeout"`
+	GracefulPeriod   time.Duration `yaml:"graceful_period"`
```
[the remark this answers is not in the document]
Comment (mapno) [reply modules/distributor/distributor_ring.go:24]: And changes to where it's referenced are needed.

**Review by mapno — COMMENTED** [review]
Thank you for contributing!

Just left two minor comments. Also, don't forget to sign the CLA, thanks.

File: modules/distributor/distributor_ring.go [hunk 3]
```diff
@@ -18,9 +18,10 @@ import (
 // is used to strip down the config to the minimum, and avoid confusion
 // to the user.
 type RingConfig struct {
-	KVStore          kv.Config     `yaml:"kvstore"`
-	HeartbeatPeriod  time.Duration `yaml:"heartbeat_period"`
-	HeartbeatTimeout time.Duration `yaml:"heartbeat_timeout"`
+	KVStore                kv.Config     `yaml:"kvstore"`
+	HeartbeatPeriod        time.Duration `yaml:"heartbeat_period"`
+	HeartbeatTimeout       time.Duration `yaml:"heartbeat_timeout"`
+	GracefulShutdownPeriod time.Duration `yaml:"graceful_shutdown_period"`
```
Comment (joe-elliott) [inline modules/distributor/distributor_ring.go:24]: we should put this on the main distributor config instead of the ring config

Comment (Joffref) [reply modules/distributor/distributor_ring.go:24]: I think that makes sense, went too fast on this one :) On it today

Comment (Joffref) [reply modules/distributor/distributor_ring.go]: It took time, but it is solved 😅

**Review by zalegrala — COMMENTED** [review]
This looks reasonable, thanks for the PR.  This might be good to document also in the main configuration page.  Please rebase for the conflict.

**Pull request closed without merging**